### ISSUE TITLE
serde::json macro invocation failed to parse

### ISSUE THREAD

**Issue by extremegf** [rust-analyzer/rust-analyzer#6876]
This is a new problem, as the snippets that now fail in my codebase used to be handled without issue.

I did some bisecting. This worked fine on v0.2.400 and fails on  v0.2.408. It also fails on  v0.2.416, albeit with a smaller red highlight.

Steps to reproduce:

lib.rs
```rust
use serde_json::json;

fn foo() {
    json!({ "a": "b" });
}
```

Cargo.toml
```toml
[dependencies]
serde = "1.0"
serde_json = "1"
```

Observe that the json! call is highlighted red with "failed to parse macro invocation".

**Comment by flodiebold** [comment]
It probably failed to expand before those versions; we just didn't have the diagnostic yet. You can disable the error for now by adding `"macro-error"` to the `rust-analyzer.diagnostics.disabled` setting.

**Comment by flodiebold** [comment]
This was also reported as a comment in #6704, but it might not be the same issue.

**Comment by lnicola** [comment]
Reproducer:

```rust
macro_rules! json_internal {
    (@object $object:ident ($($key:tt)*) ($tt:tt $($rest:tt)*)) => {};

    ({ $($tt:tt)+ }) => {
        json_internal!(@object object () ($($tt)+));
    };
}

fn main() {
    json_internal!({ "a": "b" });
}
```

**Comment by flodiebold** [comment]
Hmm that looks like the same expanding-to-statements problem as #6811 then.

**Comment by lnicola** [comment]
That's right. Closing as duplicate of #6811.

**Comment by extremegf** [comment]
> It probably failed to expand before those versions; we just didn't have the diagnostic yet. You can disable the error for now by adding `"macro-error"` to the `rust-analyzer.diagnostics.disabled` setting.

Thank you. That's a useful workaround!

I'm impressed at how quickly this was analyzed by the RA team.
